compiler/runtime/Runtime.hpp

```diff
@@ -170,31 +170,66 @@ enum TR_RuntimeHelper
 class TR_RuntimeHelperTable
    {
 public:
-   void* getAddress(TR_RuntimeHelper h)
-      {
-      return h < TR_numRuntimeHelpers ? _helpers[h] : (void *) (uintptr_t) 0xDEADB00F;
-      }
+
+   /**
+    * \brief
+    *
+    * For helper function, this returns the raw entry address. If the runtime
+    * helper's corresponding linkage is not a function, return the helper as data.
+    */
+   void* getAddress(TR_RuntimeHelper h);
+
+   /**
+    * \brief
+    *
+    * Looks up the given helper index and returns the environment pointer, if there is any.
+    * The environment pointer of a function is applicable to PPC AIX and z/OS.
+    */
```

Comment:
I suggest removing the _applicable to ... AIX and z/OS_ part because it will go stale if/when other platforms that use this convention are added. For example, Linux ppc64 also has this.

Also, I'm a little confused about what's supposed to be returned here. Is it a pointer to the function descriptor? Or a particular field (that isn't the entry point because that's what `getAddress` seems to be doing)?

The "environment pointer" in PPC ABIs is a field in the function descriptor that's mostly useless to us at the moment, so that's another source of confusion, and `TOC_UNWRAP_ENV` part that you added seems to suggest you're returning a particular field after all. Is it the TOC pointer?

Reply:
Removed the statement `applicable to ... AIX and z/OS` and updated the comment to state explicitly that this returns the `environment field extracted from the function descriptor on applicable platforms`.

Reply:
This goes without saying that the `getEnvironment()` is an empty stub in the OMR project. So it's another reason to remove the words on AIX and z/OS. The actual environment unwrapping should be project specific.

Reply:
Would it be better to only have APIs to return the function descriptor and the entry point, rather than a specific field and the entry point? That way the caller can get access to any field in the descriptor that they know and care about, rather than just one particular field. The way it is now is OK until someone decides they really want to access 2 or more fields from their descriptors or want access to the descriptor itself.

The most natural approach to me is to have `getFunctionPointer` and `getEntryPoint` and to have those return

* the function descriptor and entry point on platforms that use function descriptors, or
* the entry point and entry point on platforms that don't.

Anyone who wants to call the function from C will naturally use `getFunctionPointer`, anyone who knows about the platform ABI will know whether or not function pointers are pointers to descriptors or entry points and will use `getFunctionPointer` and know what fields are available in descriptors, and anyone who wants the actual entry point will call `getEntryPoint` without caring. The bonus is that you also have a convenient and obvious way to tell if function pointers are pointers to code by testing `getFunctionPointer() == getEntryPoint()`.

I'd also use `getEntryPoint` rather than `getAddress` because there is less room for confusion and most ABIs I've read refer to entry points.